src/tutorial/hello-world/README.md

```diff
@@ -6,73 +6,65 @@ Note: This section is a continuation of ['The Basics'](/tutorial/the-basics/). I
-0x20            000000000000000000000000000000000000000000000000000000000000000d // The length of "Hello, world!" in bytes
-0x40            0000000000000000000000000000000000000048656c6c6f2c20776f726c6421 // Wrong encoding!
-```
-This is not what we need. Instead, we're going to leverage what we know about memory to produce the right result:
-- Memory is always expanded in 32 byte increments
-- We can store a value starting from any index
-- Values are left padded
-
-Additionaly, in this example, we know that we want to store the location starting at 0x00 and the length starting at 0x20.
-
-Armed with this knowledge, we can store the value "Hello, world!" at 0x2d (starting index of length 0x20 + the length of the data in bytes 0x0d). The first 0x2d (decimal 45) bytes are set to zero as a result of memory expansion. The left padding of "Hello, world!" is stored from bytes 0x2d (decimal 45)-> 0x3f (decimal 63).
-
-This means our important bytes (`48656c6c6f2c20776f726c6421`) start at 0x40. However, since memory is expanded in 32 byte increments empty memory will be expanded with zeros to byte 0x60 (decimal 96), exactly what we need! 
-```
-[Byte number]   [DATA]    
-0x00            0000000000000000000000000000000000000000000000000000000000000000 // Empty
-0x20            0000000000000000000000000000000000000000000000000000000000000000 // Empty
-0x40            48656c6c6f2c20776f726c642100000000000000000000000000000000000000 // Correct encoding 
-```
-
-Now we have "Hello, world!" stored, we have two things left to do. Store the length of the string in 0x20, and a pointer to the start of our dynamic data 0x00! Now let's see what that looks like in code.
-
 ## Implementation
-The following `MAIN` macro steps through this encoding in a clear way (gas optimization will be left as an exercise to the reader!)
+The following `MAIN` macro steps through this encoding in a clear way.
 ```
+
 #define macro MAIN() = takes(0) returns(0) {
-    // Store string "Hello, world1" in memory at 0x40
-    // 0x2d is listed as mstore pads the value
-    0x48656c6c6f2c20776f726c6421        // ["Hello, world!"]
-    0x2d                                // [0x2d, "Hello, world!"]
+    // store dynamic offset of 0x20 at 0x00
+    0x20                                // [0x20]
+    0x00                                // [0x00, 0x20]
     mstore                              // []
 
-    // store length of string at 0x20
+    // store string length of 0x0d at 0x20
     0x0d                                // [0x0d]
     0x20                                // [0x20, 0x0d]
     mstore                              // []
-    
-    // store dynamic offset at 0x00 (string encoding starts at 0x20)
-    0x20                                // [0x20]
-    0x00                                // [0x00, 0x20]
+
+    // store bytes for "Hello, world!" at 0x40
+    0x48656c6c6f2c20776f726c642100000000000000000000000000000000000000  // <- we have to right pad this because its bytes
```

Comment:
We can use the `__RIGHTPAD()` builtin

Reply:
fixed https://github.com/huff-language/huff-docs/pull/39/commits/8c2e792cddd3320189af72ee885283118413a8fc